src/validation.cpp

```diff
@@ -2067,7 +2067,16 @@ bool ApplyTxInUndo(const CTxInUndo& undo, CCoinsViewCache& view, const COutPoint
         CCoinsModifier coins = view.ModifyCoins(out.hash);
         if (undo.nHeight != 0 ||
                 // Special-case genesis since nHeight is always 0, assume DB is clean
-                (Params().GenesisBlock().vtx[0]->GetHash() == out.hash && coins->IsPruned())) {
+                (Params().GenesisBlock().vtx[0]->GetHash() == out.hash && coins->IsPruned()) ||
+                // Special-case for initial issuance since being a single output nHeight is always 0
+                (Params().GenesisBlock().vtx.size() > 1 && [&]() {
```

Comment:
shouldn't this be ```Params().GenesisBlock().vtx.size() >= 1``` ?

Reply:
it does an identical check for the first transaction case already. It's redundant however.

Reply:
Thought it’s covered by the first clause. Should I change something?